.github/workflows/static-analysis.yaml

```diff
@@ -12,13 +12,8 @@ jobs:
       - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
         with:
           go-version: ${{ steps.goversion.outputs.goversion }}
-      - id: golangci_lint_version
-        run: echo "golangci_lint_version=$(cd tools/mod && go list -m -f {{.Version}} github.com/golangci/golangci-lint)" >> "$GITHUB_OUTPUT"
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@9d1e0624a798bb64f6c3cea93db47765312263dc # v5.1.0
-        with:
-          version: ${{ steps.golangci_lint_version.outputs.golangci_lint_version }}
-          args: --config tools/.golangci.yaml
+      - run: |
+          go install github.com/golangci/golangci-lint/cmd/golangci-lint@"$(cd tools/mod && go list -m -f {{.Version}} github.com/golangci/golangci-lint)"
```

Comment:
Doesn't golangci-lint install automatically as part of `make verify` if it's not present already?

i.e. the same way the prow job works where we don't install it automatically.  Refer:
1. https://github.com/etcd-io/etcd/blob/main/Makefile#L81
2. https://github.com/etcd-io/etcd/blob/main/Makefile#L159

Reply:
It doesn't work for GitHub actions because it tries to install it in `$GOPATH/bin`. I should have a run with an error somewhere...

Reply:
I couldn't find it, as I have many jobs with failures from experiments :sweat_smile:... So, it was quicker to force a new run. Refer to https://github.com/ivanvc/etcd/actions/runs/8963715131/job/24614385728

> golangci/golangci-lint info checking GitHub for tag 'v1.57.2'
golangci/golangci-lint info found version: 1.57.2 for v1.57.2/linux/amd64
install: cannot create regular file '/bin/golangci-lint': Permission denied
PASSES="lint" ./scripts/test.sh

For the commit when I removed the installation: https://github.com/ivanvc/etcd/commit/66b7a2bcce7cce1b495a36a55f383f707ae3d14e

Reply:
Hmm I would have expected setup-go to handle this but it seems it doesn't. There is some discussion here and some workarounds: https://github.com/actions/setup-go/issues/14 

Happy to leave it as is for now it's not a big issue.